security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -692,8 +690,9 @@ func receiveThread(con *sdsConnection, reqChannel chan *xdsapi.DiscoveryRequest,
 }
 
 func constructConnectionID(proxyID string) string {
-	id := atomic.AddInt64(&connectionNumber, 1)
-	return proxyID + "-" + strconv.FormatInt(id, 10)
+	// In istio-agent model, we consider all connections to be the same. This allows us to cache secrets
+	// across connections (by resource name).
+	return "bootstrap"
```

Comment:
Just curious, I may not follow the latest change in SDS in istio-agent model.

I remember the connection ID was added to address the issue that enovy send out multiple stream for same sdsconfig/resourcename, see https://github.com/istio/istio/pull/11738. Is that no longer an issue with this PR?

Reply:
Its still sending two connections for the same resource name (`default`) until merges https://github.com/istio/istio/pull/21108. I don't see why we need the connectionID to be unique for this case though, although perhaps there are some cases where it does matter. Let me dig into a bit more where we use connection id (maybe one of the failing tests will give a clue, I haven't fixed those yet).

Reply:
Also a big difference now is that SDS server is 1:1 with Envoy, since its run locally. So that likely is the different. But Ill look into it more

Reply:
cc @JimmyCYJ

Reply:
I think this actually will break some stuff, trying out another approach locally now